Cubical/Foundations/FinSet.agda

```diff
@@ -0,0 +1,43 @@
+{-# OPTIONS --cubical --no-import-sorts --safe #-}
+
+module Cubical.Foundations.FinSet where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Logic
+open import Cubical.Foundations.Function
+open import Cubical.Foundations.Structure
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Foundations.Univalence
+open import Cubical.Foundations.Equiv
+open import Cubical.HITs.PropositionalTruncation
+open import Cubical.Data.Unit
+open import Cubical.Data.Nat
+open import Cubical.Data.Fin
+
+isFinSet : Type₀ → Type₁
+isFinSet A = ∥ Σ[ n ∈ ℕ ] A ≡ Fin n ∥
```

Comment:
Could use `isFinSet A = Σ[ n ∈ ℕ ] ∥ A ≃ Fin n ∥` here, which would be stronger and more universe-polymorphic.

Reply:
There are two suggestions here: (1) To replace equality by equivalence (this makes the definition more universe polymorphic) and (2) to move the truncation inside the Sigma (this would make the definition stronger). But actually, regarding (2), you can prove that the definitions with truncation inside and outside are equivalent, because if `Fin n` and `Fin n'` are equivalent, then `n` and `n'` are equal, and because the natural numbers form a set.

Reply:
I agree with Martín, please use `isFinSet A = ∥ Σ[ n ∈ ℕ ] A ≃ Fin n ∥`. There is a nice notation for this in https://github.com/agda/cubical/blob/master/Cubical/Data/Sigma/Base.agda#L29

This will make it trivial to prove that the property of being finite is a proposition. You should then prove that you can obtain the version that @WorldSEnder suggested